src/System.IO.Compression.Brotli/System/IO/Compression/BrotliStream.cs

```diff
@@ -346,27 +315,19 @@ public override void Write(byte[] buffer, int offset, int count)
                     throw new TimeoutException(BrotliEx.TimeoutWrite);
                 }
                 copyLen = bytesRemain > _bufferSize ? _bufferSize : bytesRemain;
-                Marshal.Copy(buffer, currentOffset, _bufferInput, copyLen);
+                byte[] bufferInput = new byte[copyLen];
```

Comment:
Why is this extra buffer allocation and copy necessary? If you are going to need an extra buffer on each write call, you might want to rent a buffer from a pool, have your own buffer that you re-use, or use a stack buffer (if it isn't too large).